src/utils/googleTagManager.ts

```diff
@@ -155,14 +166,64 @@ export const trackEvent = ({
   action: string
   label?: EventLabel
 }): void => {
-  const dataLayer = {
+  const dataLayer: EventDataLayer = {
     event,
     chainId: _getChainId(),
     eventCategory: category,
     eventAction: action,
     eventLabel: tryParse(label),
   }
 
+  track(dataLayer)
+}
+
+type SafeAppEventDataLayer = {
+  event: GTM_EVENT.SAFE_APP
+  chainId: string
+  safeAppName: string
+  safeAppMethod: string
+  safeAppEthMethod?: string
+  safeAppSDKVersion?: string
+}
+
+export const getSafeAppName = (safeApp?: SafeApp): string => {
+  if (!safeApp?.id) {
+    return EMPTY_SAFE_APP
+  }
+
+  try {
+    const parsedSafeApp = JSON.parse(safeApp.id)
+
+    return parsedSafeApp.name || parsedSafeApp.url || parsedSafeApp
```

Comment:
If it falls back to `parsedSafeApp`, could that be an object? Shouldn't this always be a string?

Reply:
right, removed the fallback